src/Extensions/UserDefinedFormControllerExtension.php

```diff
@@ -2,95 +2,108 @@
 
 namespace SilverStripe\SmimeForms\Extensions;
 
-use Exception;
 use Psr\Container\NotFoundExceptionInterface;
 use SilverStripe\Assets\Dev\TestAssetStore;
 use SilverStripe\Assets\File;
 use SilverStripe\Assets\Flysystem\FlysystemAssetStore;
 use SilverStripe\Assets\Storage\AssetStore;
 use SilverStripe\Control\Email\Email;
-use SilverStripe\Control\Email\Mailer;
 use SilverStripe\Core\Injector\Injector;
 use SilverStripe\ORM\DataExtension;
-use SilverStripe\SMIME\Control\SMIMEMailer;
+use SilverStripe\SmimeForms\Model\SmimeEmail;
 use SilverStripe\SmimeForms\Model\SmimeEncryptionCertificate;
 use SilverStripe\SmimeForms\Model\SmimeSigningCertificate;
 use SilverStripe\UserForms\Model\Recipient\EmailRecipient;
 
 /**
  * Class UserDefinedFormControllerExtension
  *
- * An extension for {@see UserDefinedFormController} class to check whether the form submission needs
- * to be encrypted. If so, it will replace the standard Mailer with a {@see SMIMEMailer}.
+ * An extension for {@see UserDefinedFormController} class to check whether the
+ * form submission needs to be encrypted. If so, this will use the SmimeEmail
+ * class to both sign and encrypt the email prior to sending it using symfony/Mailer.
  *
  * @package SilverStripe\SmimeForms\Extensions
  */
 class UserDefinedFormControllerExtension extends DataExtension
 {
 
     /**
-     * Called as an extension hook from {@see UserDefinedFormController}.
+     * Uses Injector to fully override the Email class with
+     * the SMimeEmail class if a form enables email encryption.
      *
+     * We use the 'updateEmailData' hook to do this as the Email
+     * instance is yet to be created at this point of the process.
+     *
+     * Meaning once this hook is executed the next Email instance
+     * that is created will be of type SMimeEmail.
+     *
+     * @param array $emailData
+     * @param array $attachments
+     * @return void
+     */
+    public function updateEmailData(array $emailData, array $attachments): void
+    {
+        // Early exit if this form should not be encrypted
+        if (!$this->owner->isEncryptEmail()) {
+            return;
+        }
+
+        // Otherwise, use Injector to fully override Email with SMimeEmail
+        Injector::inst()->registerService(new SmimeEmail(), Email::class);
```

Comment:
CORE: Only transform into a `SmimeEmail` if the encryption option is enabled so it doesn't apply to every single`Email` instance.